src/main/java/digital/slovensko/autogram/core/eforms/XDCValidator.java

```diff
@@ -0,0 +1,105 @@
+    private static final Charset ENCODING = StandardCharsets.UTF_8;
+
+    public static boolean isXDCContent(DSSDocument document) {
+        try {
+            var doc = getXmlFromDocument(document);
+            var docString = transformElementToString(doc.getDocumentElement());
+            var xdcSchema = EFormUtils.class.getResourceAsStream("xmldatacontainer.xsd");
+
+            return validateXmlContentAgainstXsd(docString, new String(xdcSchema.readAllBytes(), ENCODING));
+
+        } catch (IOException | NullPointerException e) {
+            return false;
+        }
+    }
+
+    public static void validateXml(String xsd, String xslt, DSSDocument xmlDocument, String cannonicalizationMethod, DigestAlgorithm digestAlgorithm)
+            throws OriginalDocumentNotFoundException, XMLValidationException, XMLValidationException {
+        if (xmlDocument == null)
+            throw new XMLValidationException("Zlyhala validácia XML Datacontainera", "Nepodarilo sa načítať XML dokument");
+
+        var xml = EFormUtils.getXmlFromDocument(xmlDocument);
+        if (xml == null)
+            throw new XMLValidationException("Zlyhala validácia XML Datacontainera", "Nepodarilo sa načítať XML dáta z XML Datacontainera");
+
+        if (AutogramMimeType.isXDC(xmlDocument.getMimeType())) {
+            if (!XDCValidator.isXDCContent(xmlDocument))
+                throw new XMLValidationException("Zlyhala validácia XML Datacontainera",
+                        "Poskytnutý XML dokument nie je validný XML Datacontainer podľa XSD schémy");
+
+            if (xsd != null && !XDCValidator.validateXsdDigest(xsd, xml.getDocumentElement(), cannonicalizationMethod, digestAlgorithm))
+                throw new XMLValidationException("Zlyhala validácia XML Datacontainera", "XSD schéma sa nezhoduje s odtlačkom v XML Datacontaineri");
+
+            if (xslt != null && !XDCValidator.validateXsltDigest(xslt, xml.getDocumentElement(), cannonicalizationMethod, digestAlgorithm))
+                throw new XMLValidationException("Zlyhala validácia XML Datacontainera", "XSLT transformácia sa nezhoduje s odtlačkom v XML Datacontaineri");
+        }
+
+        var eformContent = EFormUtils.transformElementToString(AutogramMimeType.isXDC(xmlDocument.getMimeType())
+                ? EFormUtils.getEformXmlFromXdcDocument(xmlDocument).getDocumentElement()
+                : xml.getDocumentElement());
+
+        if (xsd != null && !validateXmlContentAgainstXsd(eformContent, xsd))
+            throw new XMLValidationException("Zlyhala validícia XML dokumentu", "XML dokument sa nezhoduje s XSD schémou");
+    }
+
+    public static boolean validateXmlContentAgainstXsd(String xmlContent, String xsdSchema) {
+        if (xsdSchema == null)
+            return true;
+
+        try {
+            var factory = SchemaFactory.newInstance(XMLConstants.W3C_XML_SCHEMA_NS_URI);
+            factory.setFeature(XMLConstants.FEATURE_SECURE_PROCESSING, true);
```

Comment:
Tu nie je problem ze nie je vypnute to parsovanie externych entit?

Reply:
Je to iné. SchemaFactory nemá také veci a keď som tam dal takú feature, tak to padlo. Ten spôsob vypínania EE funguje pre DocumentBuilderFactory